### PR TITLE
Rewrite ynh-dev

### PR DESCRIPTION
**Pull request by alexAubin**
Alright guyz, I'm making this PR for (basically) a rewrite of ynh-dev. As for #26, yes, it is not "easily reviewable" because too many things changed and I honestly don't have the energy to take 4+ hours to correctly split this rewriting into separate pieces considering it's too big and there's only a handful of people who actually use this internal tool. I'm ready to maintain my own fork if people don't want to merge this.

Trying to summarize the motivation and what was done in this rewriting : 
- [X] Based on @zamentur [LXC integration](https://github.com/YunoHost/ynh-dev/tree/enh-vagrant-lxc)
- [X] Refactor with ~defensive bash + better argument parsing
- [X] Drop support of Virtualbox, too much of a burden and less efficient than LXC
- [X] Merge the [VagrantFile repo](https://github.com/YunoHost/VagrantFile) directly inside ynh-dev, no need to have so many different repos...
- [X] Remove 'create-env', have a script to deploy a new `ynh-dev` folder instead. The mid-term idea being to be able to `curl https://install.yunohost.org/dev | bash` (though need to take care of the dependencies).
- [X] Remove the concept of 'domain' (or rather box name) and 'versions' of box which is overly complicated and confusing for new comers. Instead, have a single and simple `./ynh-dev start` which just start the default box. (Advanced users can still define additional work containers if they need to have several containers running at the same time)
- [X] Add a `ssh` action to SSH into the machine (newcomers don't know about `vagrant ssh` and what's the name of the box)
- [X] Add a `rebuild` command to rebuild the LXC from ~scratch locally. Shouldn't have to guess that this stuff is hidden in VagrantFile/prebuild
- [X] Removed some commands which I deemed useless
- [X] Update / try to simplify README
- [X] Misc tweaks for UX

### CODE REVIEW

File: ynh-dev [hunk 1]
```diff
@@ -1,382 +1,306 @@
...
-                cd /vagrant/yunohost/src/yunohost
+                cd /ynh-dev/yunohost/src/yunohost
                 py.test tests
                 ;;
         esac
     done
+}
 
-
-elif [ "$1" = "self-update" ]; then
```
Comment (zamentur) [inline ynh-dev]: Do you voluntarily remove this command ?

Comment (alexAubin) [reply ynh-dev]: Yes ... I don't feel like it's worth creating a command for it. It just does a "git pull" (and inside the vm, which is weird ?)

**Comment by zamentur** [comment]
If this pr is accepted we could close this other one https://github.com/YunoHost/ynh-dev/pull/21

**Comment by alexAubin** [comment]
Any opinions on this ?

**Pull request merged**